examples/gnrc_tftp/tftp_client.c

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (C) 2015 Engineering-Spirit
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file
+ * @brief       Demonstrating the sending and receiving of data via the TFTP client
+ *
+ * @author      Nick van IJzendoorn <nijzendoorn@engineering-spirit.nl>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <inttypes.h>
+
+#include "kernel.h"
+#include "net/gnrc/tftp.h"
+
+static const char* _tftp_default_host = "::1";
+
+/* default server text which can be received */
+static const char _tftp_client_hello[] = "Hello,\n"
+        "\n"
+        "Client text would also need to exist to be able to put data.\n"
+        "\n"
+        "Enjoy the RIOT-OS\n";
+
+static tftp_action_t _tftp_action;
+
+/**
+ * @brief called at every transaction start
+ */
+static bool _tftp_client_start_cb(tftp_action_t action, tftp_mode_t mode,
+                                  const char *file_name, size_t *len) {
+    /* translate the mode */
+    const char *str_mode = "ascii";
+    if (mode == TTM_OCTET)
+        str_mode = "bin";
+    else if (mode == TTM_MAIL)
+        str_mode = "main";
+
+    /* translate the action */
+    const char *str_action = "read";
+    if (action == TFTP_WRITE)
+        str_action = "write";
+
+    /* display the action being performed */
+    printf("tftp_client: %s %s %s:%u\n", str_mode, str_action, file_name, *len);
+
+    /* return the length of the text, if this is an read action */
+    if (action == TFTP_READ)
+        *len = sizeof(_tftp_client_hello);
+
+    /* remember the action of the current transfer */
+    _tftp_action = action;
+
+    /* we accept the transfer to take place so we return true */
+    return true;
+}
+
+/**
+ * @brief called to get or put data, depending on the mode received by `_tftp_start_cb(action, ...)`
+ */
+static int _tftp_client_data_cb(uint32_t offset, void *data, size_t data_len) {
+    char *c = (char*) data;
+
+    /* if we are reading return the part of the data that is being requested */
+    if (_tftp_action == TFTP_WRITE)
+    {
+        /* calculate the length of the data block to transfer */
+        if (offset + data_len > sizeof(_tftp_client_hello))
+            data_len -= (offset + data_len) - sizeof(_tftp_client_hello);
+
+        /* copy the block to the output buffer */
+        memcpy(data, _tftp_client_hello + offset, data_len);
+
+        /* return the length of the data block */
+        return data_len;
+    }
+    else
+        /* we received a data block which we output to the console */
+        return printf("\n -- CLIENT DATA --\n%*s\n -- CLIENT DATA --\n", data_len, c) - (sizeof("\n -- CLIENT DATA --\n%*s\n -- CLIENT DATA --\n") - 4);
```

Comment:
I don't really understand this line. Why don't you just return `data_len`? And please add curly braces ([RIOT coding conventions](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions#indentation-and-braces)).